### PR TITLE
update honduras portal bulk spider

### PR DESCRIPTION
**Pull request by Ravf95**
Ref #414

### CODE REVIEW

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py [hunk 1]
```diff
@@ -41,39 +45,48 @@ class HondurasPortalBulk(PeriodicSpider):
...
-    available_systems = {'HC1': 'honducompras-1', 'CE': 'catalogo-electronico', 'DDC': 'difusion-directa-contrato'}
+    def start_requests(self):
+        url = 'http://www.contratacionesabiertas.gob.hn/api/v1/descargas/?format=json'
+        yield scrapy.Request(url, meta={'file_name': 'list.json'}, callback=self.parse_list)
 
     @classmethod
     def from_crawler(cls, crawler, publisher=None, system=None, *args, **kwargs):
         spider = super().from_crawler(crawler, publisher=publisher, system=system, *args, **kwargs)
-        if publisher and spider.publisher not in spider.available_publishers:
+        if publisher and spider.publisher not in spider.available_publishers.keys():
```
Comment (yolile) [inline kingfisher_scrapy/spiders/honduras_portal_bulk.py:61]:
```suggestion
        if publisher and spider.publisher not in spider.available_publishers:
```
.keys() is the default behavior

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py [hunk 2]
```diff
@@ -41,39 +45,48 @@ class HondurasPortalBulk(PeriodicSpider):
...
             raise SpiderArgumentError(f'spider argument `publisher`: {spider.publisher!r} not recognized')
 
         if system:
             if spider.publisher != 'oncae':
                 raise SpiderArgumentError(f'spider argument `system` is not supported for publisher: '
                                           f'{spider.publisher!r}')
-            if spider.system not in spider.available_systems:
+            if spider.system not in spider.available_systems.keys():
```
Comment (yolile) [inline kingfisher_scrapy/spiders/honduras_portal_bulk.py:68]:
```suggestion
            if spider.system not in spider.available_systems:
```

.keys() is the default behavior

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py [hunk 3]
```diff
@@ -41,39 +45,48 @@ class HondurasPortalBulk(PeriodicSpider):
...
-                                              f"{self.available_systems[system]}_{date.year}_{date.month:02d}.json")
-            else:
-                yield self.pattern.format(f"{self.available_publishers[publisher]}_"
-                                          f"{date.year}_{date.month:02d}.json")
+            if publisher == self.available_publishers['oncae']:
+                system = item['sistema']
+                if self.system and system != self.available_systems.get(self.system):
+                    continue
+
+            date = datetime(int(item['year']), int(item['month']), 1)
```
Comment (yolile) [inline kingfisher_scrapy/spiders/honduras_portal_bulk.py:87]: You can move this inside the if block, as the date is only needed if the from_date and until_date are set

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py [hunk 4]
```diff
@@ -41,39 +45,48 @@ class HondurasPortalBulk(PeriodicSpider):
...
         return spider
 
-    def build_urls(self, date):
-        for publisher in self.available_publishers:
-            if self.publisher and publisher != self.publisher:
+    @handle_http_error
+    def parse_list(self, response):
+        formatter = components(-1)
+
+        for item in response.json():
```
Comment (yolile) [inline kingfisher_scrapy/spiders/honduras_portal_bulk.py:77]: Could you add a comment on how each item looks so it is easier for another developer to understand the code and filters? eg 
``` python
# "Each item looks like:  {
#    "urls": {
#      "csv": "...",
#      "md5": "...",
#      "json": "...",
#      "xlsx": "..."
#    },
#    "year": "2012",
#    "month": "01",
#    "sistema": "Sistema de Informacion Financiero Integrado v2.0",
#    "publicador": "Secretaria de Finanzas de Honduras"
#  }

**Review by yolile — CHANGES_REQUESTED** [review]

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py (hunk 1 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/honduras_portal_bulk.py:61]: Yes, you're right!. I will remove it on the next commit

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py (hunk 2 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/honduras_portal_bulk.py:68]: Yes, you're right!. I will remove it on the next commit

File: kingfisher_scrapy/spiders/honduras_portal_bulk.py (hunk 4 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/honduras_portal_bulk.py:77]: Yes, I will add a comment

**Review by yolile — APPROVED** [review]

**Pull request merged**